crates/nargo_cli/tests/test_data/higher_order_fn_selector/src/main.nr

```diff
@@ -0,0 +1,49 @@
+use dep::std;
+
+fn f(x: &mut Field) -> Field {
+    *x = *x - 1;
+    1
+}
+
+fn g(x: &mut Field) -> Field {
+    *x *= 2;
+    1
+}
+
+fn h(x: &mut Field) -> Field {
+    *x *= 3;
+    1
+}
+
+fn selector(flag:&mut bool) -> fn(&mut Field) -> Field { //TODO: Can we have fn(&mut Field) -> () return type?
```

Comment:
You should be able to, is there a bug?

Reply:
huh, I might have been confused, i improved the test now and used that